src/main/java/org/opentripplanner/middleware/auth/RequestingUser.java

```diff
@@ -36,33 +36,42 @@ private Auth0UserProfile(String auth0UserId) {
             this.auth0UserId = auth0UserId;
             Bson withAuth0UserId = eq("auth0UserId", auth0UserId);
             otpUser = Persistence.otpUsers.getOneFiltered(withAuth0UserId);
-            adminUser = Persistence.adminUsers.getOneFiltered(withAuth0UserId);
             apiUser = Persistence.apiUsers.getOneFiltered(withAuth0UserId);
+            adminUser = Persistence.adminUsers.getOneFiltered(withAuth0UserId);
         }
     }
 
     /**
-     * Create a user profile from the request's JSON web token. Check persistence for stored user
+     * Create a user profile from the request's JSON web token. Check persistence for stored user.
      */
-    public Auth0UserProfile(DecodedJWT jwt) {
+    public RequestingUser(DecodedJWT jwt) {
         this.auth0UserId = jwt.getClaim("sub").asString();
         Bson withAuth0UserId = eq("auth0UserId", auth0UserId);
         otpUser = Persistence.otpUsers.getOneFiltered(withAuth0UserId);
-        adminUser = Persistence.adminUsers.getOneFiltered(withAuth0UserId);
         apiUser = Persistence.apiUsers.getOneFiltered(withAuth0UserId);
+        adminUser = Persistence.adminUsers.getOneFiltered(withAuth0UserId);
     }
 
     /**
      * Utility method for creating a test user. If a Auth0 user Id is defined within the Authorization header param
      * define test user based on this.
      */
-    static Auth0UserProfile createTestUser(Request req) {
+    static RequestingUser createTestUser(Request req) {
         String auth0UserId = null;
+
         if (isAuthHeaderPresent(req)) {
             // If the auth header has been provided get the Auth0 user id from it. This is different from normal
             // operation as the parameter will only contain the Auth0 user id and not "Bearer token".
             auth0UserId = req.headers("Authorization");
         }
-        return new Auth0UserProfile(auth0UserId);
+
+        return new RequestingUser(auth0UserId);
+    }
+
+    /**
+     * Determine if requesting user is a third party user.
+     */
+    public boolean isThirdPartyUser() {
```

Comment:
Suggestion: rename to `isThirdParty` (so the code reads like `requestingUser.isThirdParty()`)

We probably need a similar `isAdmin` method too for anywhere we check for admin rights.

Reply:
Do we need the comment above?

Reply:
Renamed method. No harm in keeping the comment is there?

Reply:
The comment should be augmented in my view. `isThirdParty` means very little if you're just reading the code and are not privy to our team discussions. The comment could distinguish between "first party" and probably have a TODO about enhancing this check as the application develops further (perhaps adding/checking a specific field on ApiUser that indicates this).

Reply:
Updated.

Reply:
I still think this comment leaves a bit to be desired. And for the sake of wiping out any ambiguity, I think the method should be called `isThirdPartyApiUser` or `isThirdPartyApplication`. My suggestion:

> Determine if the requesting user is a third party API user. A third party API user is classed as a user that has signed up for access to the otp-middleware API. These users are expected to make requests on behalf of the OtpUsers they sign up, presumably via a server application that authenticates via otp-middleware's authenticate endpoint. OtpUsers created for third party API users enjoy a more limited range of activities (e.g., they cannot receive email/SMS notifications from otp-middleware).